recipes/jhs.rb

```diff
@@ -60,86 +60,62 @@
   deps += "namenode.service "
 end
 
-if node['hops']['systemd'] == "true"
-
-  service service_name do
-    provider Chef::Provider::Service::Systemd
-    supports :restart => true, :stop => true, :start => true, :status => true
-    action :nothing
-  end
-
-  case node['platform_family']
-  when "debian"
-    systemd_script = "/lib/systemd/system/#{service_name}.service"
-  when "rhel"
-    systemd_script = "/usr/lib/systemd/system/#{service_name}.service" 
-  end
-
-
-  file systemd_script do
-    action :delete
-    ignore_failure true
-  end
-  
-  template systemd_script do
-    source "#{service_name}.service.erb"
-    owner "root"
-    group "root"
-    mode 0664
-    variables({
-      :deps => deps,
-      :nn_rpc_endpoint => consul_helper.get_service_fqdn("namenode")
-    })
-    if node['services']['enabled'] == "true"
-      notifies :enable, resources(:service => service_name)
-    end
-    notifies :restart, resources(:service => service_name)
-  end
+service service_name do
+  provider Chef::Provider::Service::Systemd
+  supports :restart => true, :stop => true, :start => true, :status => true
+  action :nothing
+end
 
-  kagent_config "#{service_name}" do
-    action :systemd_reload
-  end
-  
-  directory "/etc/systemd/system/#{service_name}.service.d" do
-    owner "root"
-    group "root"
-    mode "755"
-    action :create
-  end
+case node['platform_family']
+when "debian"
+  systemd_script = "/lib/systemd/system/#{service_name}.service"
+when "rhel"
+  systemd_script = "/usr/lib/systemd/system/#{service_name}.service" 
+end
 
-  template "/etc/systemd/system/#{service_name}.service.d/limits.conf" do
-    source "limits.conf.erb"
-    owner "root"
-    mode 0774
-    action :create
-  end 
 
-else #sysv
+file systemd_script do
+  action :delete
+  ignore_failure true
+end
 
-  service service_name do
-    provider Chef::Provider::Service::Init::Debian
-    supports :restart => true, :stop => true, :start => true, :status => true
-    action :nothing
+template systemd_script do
+  source "#{service_name}.service.erb"
+  owner "root"
+  group "root"
+  mode 0664
+  variables({
+    :deps => deps,
+    :nn_rpc_endpoint => consul_helper.get_service_fqdn("namenode")
+  })
+  if node['services']['enabled'] == "true"
+    notifies :enable, resources(:service => service_name)
   end
+  notifies :restart, resources(:service => service_name)
+end
 
-  template "/etc/init.d/#{service_name}" do
-    source "#{service_name}.erb"
-    owner "root"
-    group "root"
-    mode 0755    
-if node['services']['enabled'] == "true"
-    notifies :enable, resources(:service => service_name)
+kagent_config "#{service_name}" do
+  action :systemd_reload
 end
-    notifies :restart, resources(:service => service_name), :immediately
-  end
 
+directory "/etc/systemd/system/#{service_name}.service.d" do
+  owner "root"
+  group "root"
+  mode "755"
+  action :create
 end
 
+template "/etc/systemd/system/#{service_name}.service.d/limits.conf" do
```

Comment:
That one should be moved before restarting the service above or send a notification to restart it, unless if we restart it later. Otherwise limits won't have any affect

Reply:
Actually these don't make any sense as they are already in the systemd unit.